src/Migrations/M044MigrateArticlesBlockToPostsListBlock.php

```diff
@@ -0,0 +1,367 @@
+        // add "Additional CSS class(es)" to new migrated block, if exists.
+        $additional_class = $existing_block_attrs['additional_class'];
+        if ($additional_class) {
+            $classname = $classname . ' ' . $additional_class;
+        }
+
+        $attrs = self::set_query_block_attrs($tags, $posts_override, $post_types, $layout_type, $rows, $current_post_id);
+
+        $inner_blocks = [
+            self::get_head_group_block($existing_block_attrs['title']),
+            self::get_paragraph_block($existing_block_attrs['description']),
+            self::get_query_no_results_block(),
+            self::get_post_template(),
+            self::get_buttons_block(),
+            self::get_nav_links_block(),
+        ];
+
+        return Utils\Functions::create_block_query(
+            $inner_blocks,
+            $attrs,
+            $classname
+        );
+    }
+
+    /**
+     * Get the attributes of a Post lists block.
+     *
+     * @param array $existing_block - The current Articles block.
+     * @return array - The attributes.
+     */
+    private static function get_posts_list_block_attrs(array $existing_block): array
+    {
+        $attrs = $existing_block['attrs'];
+
+        return [
+            'title' => isset($attrs['article_heading']) && !empty($attrs['article_heading']) ?
+                $attrs['article_heading'] : __('Related Articles', 'planet4-blocks'),
+            'description' => isset($attrs['articles_description']) ? $attrs['articles_description'] : '',
+            'tags' => isset($attrs['tags']) ? $attrs['tags'] : [],
+            'posts' => isset($attrs['posts']) ? $attrs['posts'] : [],
+            'post_types' => isset($attrs['post_types']) ? $attrs['post_types'] : [],
+            'rows' => isset($attrs['article_count']) ? $attrs['article_count'] : 3,
+            'current_post_id' => isset($attrs['current_post_id']) ? $attrs['current_post_id'] : 0,
+            'additional_class' => isset($attrs['className']) ? $attrs['className'] : '',
+        ];
+    }
+
+    /**
+     * Set all the attributes to create a new Query block.
+     *
+     * @param array $tags - The list of post tags.
+     * @param array $posts_override - The list of posts to include in the query.
+     * @param array $post_types - The list of terms of the "p4-page-type" taxonomy.
+     * @param string $layout_type - The layout type (grid or flex).
+     * @param int $per_page - The number of elements per page.
+     * @param int $current_post_id - The current post ID.
+     * @return array - The attributes.
+     */
+    private static function set_query_block_attrs(
+        array $tags,
+        array $posts_override,
+        array $post_types,
+        string $layout_type,
+        int $per_page,
+        int $current_post_id
+    ): array {
+        $query = [];
+        $query['pages'] = 0;
+        $query['offset'] = 0;
+        $query['postType'] = Utils\Constants::POST_TYPES_POST;
```

Comment:
I see in the Articles Block settings we have a Post Type setting, why don't we check if it contains a value before defaulting to this.

Reply:
The articles block filters only `PostType Post` and not pages or actions, hence we have this value.